src/masking/__benches__/types.bench.ts

```diff
@@ -464,3 +464,37 @@ test("base type, multiple fragments on sub-types", (prefix) => {
     }>();
   });
 });
+
+test("does not detect `$fragmentRefs` if type contains `any`", (prefix) => {
+  interface Source {
+    foo: { bar: any[] };
+  }
+
+  bench(prefix + "instantiations", () => {
+    return {} as MaybeMasked<Source>;
+  }).types([6, "instantiations"]);
+
+  bench(prefix + "functionality", () => {
+    const x = {} as MaybeMasked<Source>;
+
+    expectTypeOf(x).branded.toEqualTypeOf<Source>();
+  });
+});
```

Comment:
This test would also succeed without this PR. This adjustment changes that:

```suggestion
test("does not detect `$fragmentRefs` if type contains `any`", (prefix) => {
  interface Source {
    foo: { bar: any[] };
    " $fragmentName": "foo";
  }

  bench(prefix + "instantiations", () => {
    return {} as MaybeMasked<Source>;
  }).types([6, "instantiations"]);

  bench(prefix + "functionality", () => {
    const x = {} as MaybeMasked<Source>;

    expectTypeOf(x).branded.toEqualTypeOf<Source>();
  });
});
```